ts/features/idpay/initiative/configuration/components/InstrumentEnrollmentSwitch.tsx

```diff
@@ -27,142 +35,146 @@ export type InstrumentEnrollmentSwitchRef = {
-    ) {
-      return (
-        <Badge style={styles.badge}>
-          <LabelSmall color="white">
-            {instrumentStatusLabels[status]}
-          </LabelSmall>
-        </Badge>
-      );
+  const renderSwitch = () => {
+    if (pot.isSome(instrumentStatus)) {
+      const status = instrumentStatus.value;
+
+      if (
+        status === InstrumentStatusEnum.PENDING_ENROLLMENT_REQUEST ||
+        status === InstrumentStatusEnum.PENDING_DEACTIVATION_REQUEST
+      ) {
+        return (
+          <NBBadge style={styles.badge}>
+            <LabelSmall color="white">
+              {instrumentStatusLabels[status]}
+            </LabelSmall>
+          </NBBadge>
+        );
+      }
     }
 
-    return (
-      <Switch
-        value={switchStatus}
-        onChange={handleChange}
-        disabled={isDisabled}
-      />
+    const switchValue = pot.map(
+      instrumentStatus,
+      status => status === InstrumentStatusEnum.ACTIVE || isEnrolling
     );
+
+    return <RemoteSwitch value={switchValue} onValueChange={handleChange} />;
   };
 
-  const instrumentInfo = pipe(
-    getPaymentMethodInfo(wallet),
-    O.getOrElse(() => ({
-      logo: <View />,
-      maskedPan: ""
-    }))
-  );
+  const instrumentLogo = getPaymentMethodLogo(wallet);
+  const instrumentMaskedPan = getPaymentMaskedPan(wallet);
 
   return (
-    <ListItem>
-      <View style={styles.listItemContainer}>
-        <View style={styles.logoAndPanContainer}>
-          {instrumentInfo.logo}
+    <NBListItem>
+      <View style={[IOStyles.flex, IOStyles.rowSpaceBetween]}>
+        <View style={styles.instrumentsInfo}>
+          {instrumentLogo}
           <HSpacer size={8} />
-          <H4>{instrumentInfo.maskedPan}</H4>
+          <H4>{`•••• ${instrumentMaskedPan}`}</H4>
         </View>
-        {renderControl()}
+        {renderSwitch()}
       </View>
-    </ListItem>
+    </NBListItem>
   );
-});
+};
+
+export const cardLogos: {
+  [key in CreditCardType]: IOLogoPaymentType | undefined;
+} = {
+  MASTERCARD: "mastercard",
+  VISA: "visa",
+  AMEX: "amex",
+  DINERS: "diners",
+  MAESTRO: "maestro",
+  VISAELECTRON: "visa",
+  POSTEPAY: "postepay",
+  UNIONPAY: "unionPay",
+  DISCOVER: "discover",
+  JCB: "jcb",
+  JCB15: "jcb",
+  UNKNOWN: undefined
+};
+
+const getPaymentMaskedPan = (wallet: Wallet): string => {
+  switch (wallet.type) {
+    case "CREDIT_CARD":
+      return wallet.creditCard?.pan ?? "";
+    default:
+      return "";
+  }
+};
+
+const getPaymentMethodLogo = (wallet: Wallet): JSX.Element => {
+  switch (wallet.type) {
+    case "CREDIT_CARD":
+      const creditCardType = CreditCardType.decode(
+        wallet.creditCard?.brand?.toUpperCase()
+      );
+
+      const logo =
+        cardLogos[
+          pipe(
+            creditCardType,
+            E.getOrElseW(() => "UNKNOWN" as const)
+          )
+        ];
+
+      if (logo !== undefined) {
+        return <LogoPayment name={logo} size={25} />;
```

Comment:
```js
 logo === undefined ? (
              <Icon name="creditCard" size={24} color={"bluegrey"} />
            ) : (
              <LogoPayment name={logo} size={24} />
            ),

```

Reply:
also, size={24} more strictly follows design standards

Reply:
> ```js
>  logo === undefined ? (
>               <Icon name="creditCard" size={24} color={"bluegrey"} />
>             ) : (
>               <LogoPayment name={logo} size={24} />
>             ),
> ```

I'd prefer to leave the logic inside the function with the UI only expecting a component to render.  More simple to read and see what's going on imho.